okhttp/src/test/java/io/grpc/okhttp/OkHttpClientTransportTest.java

```diff
@@ -1866,7 +1866,8 @@ public void proxy_200() throws Exception {
         INITIAL_WINDOW_SIZE,
         HttpConnectProxiedSocketAddress.newBuilder()
             .setTargetAddress(targetAddress)
-            .setProxyAddress(serverSocket.getLocalSocketAddress()).build(),
+            .setProxyAddress(new InetSocketAddress("localhost", serverSocket.getLocalPort()))
```

Comment:
Instead of "localhost" can we just use "127.0.0.1"? I did a small test where I modified my /etc/hosts file to have
```
192.16.10.23       localhost
```

and `localhost` resolved to that IP address which will break this test.

And can we also create the `ServerSocket` to bind to `127.0.0.1` instead of `0.0.0.0`? I don't see any upside to using `0.0.0.0`.

Reply:
> any upside to using 0.0.0.0

I feel binding on 0.0.0.0 is more close to real-world use case than 127.0.0.1. I’m ok with 127.0.0.1 if you have a reason for 127.0.0.1.

Reply:
Changed "localhost" to "127.0.0.1". Thank you for input.

Reply:
> I feel binding on 0.0.0.0 is more close to real-world use case than 127.0.0.1. I’m ok with 127.0.0.1 if you have a reason for 127.0.0.1.

Since 0.0.0.0 involves binding to all local interfaces if one of them fails I suspect the ServerSocket construction itself might fail so one more way to avoid extraneous test failures. You can look into it but I am okay with the current one.

Reply:
@sanjaypujare, we tend to prefer "localhost" because it works on IPv4-only and IPv6-only machines, although IPv4-only machines for loopback are rare so we'd favor ::1 if we had to choose an IP (but it [has bitten us before](https://github.com/grpc/grpc-java/commit/f9f603570d20d4507f4d90a3dc53915ac7db4d33)). This actually needs to be localhost here, or it will fail in our internal testing infrastructure since they are IPv6-only.

Reply:
I went ahead and deleted the most recent commit from the branch, so it is now back to "localhost"

Reply:
Oh, okay. Good to know